### PR TITLE
Add integration command name and script name validation

### PR DESCRIPTION
**Pull request by kamalq97**
## Related Issues
fixes: [link to Jira issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-12466)

## Description
Add validation to ensure command name in integrations and scripts does not begin with a digit.

### CODE REVIEW

File: .changelog/4722.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+changes:
+- description: Added new BA128 validation which check names if command names start with a digit.
```
Comment (YuvHayun) [inline .changelog/4722.yml:2]:
```suggestion
- description: Added new BA128 validation which ensures script and integration command names don't start with a digit.
```

File: demisto_sdk/commands/validate/sdk_validation_config.toml [hunk 2]
```diff
@@ -158,6 +158,7 @@ select = [
   "BA118",
   "BA119",
   "BA126",
+  "BA128",
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/sdk_validation_config.toml:161]: should also put it under 'use_git' section, we want to validate this during pr builds & nightlies.

File: demisto_sdk/commands/validate/validators/BA_validators/BA128_is_command_or_script_name_starts_with_digit.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    ValidationResult,
+)
+
+ContentTypes = Union[Integration, Script]
+
+
+class IsCommandOrScriptNameStartsWithDigitValidator(BaseValidator[ContentTypes]):
+    error_code = "BA128"
+    description = "Integration command names and script names cannot start with a digit"
+    rationale = "Not supported by the platform"
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/BA_validators/BA128_is_command_or_script_name_starts_with_digit.py:18]:
```suggestion
    rationale = "Ensure we don't add commands which are not supported by the platform."
```

File: demisto_sdk/commands/validate/validators/BA_validators/BA128_is_command_or_script_name_starts_with_digit.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    BaseValidator,
+    ValidationResult,
+)
+
+ContentTypes = Union[Integration, Script]
+
+
+class IsCommandOrScriptNameStartsWithDigitValidator(BaseValidator[ContentTypes]):
+    error_code = "BA128"
+    description = "Integration command names and script names cannot start with a digit"
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/BA_validators/BA128_is_command_or_script_name_starts_with_digit.py:17]:
```suggestion
    description = "Ensure that integration command names and script names cannot start with a digit."
```

File: demisto_sdk/commands/validate/tests/BA_validators_test.py [hunk 5]
```diff
@@ -2418,3 +2421,73 @@ def test_IsTestsSectionValidValidator_obtain_invalid_content_items():
...
+def test_is_command_or_script_name_starts_with_digit_valid():
+    """
+    Given
+    - One valid integration with one command starting with a letter character.
+    When
+    - Calling the IsCommandOrScriptNameStartsWithDigitValidator obtain_invalid_content_items function.
+    Then
+    - Make sure no failures are returned.
+    """
+    with ChangeCWD(REPO.path):
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/tests/BA_validators_test.py:2471]: not sure you need the ChangeCWD part.

File: demisto_sdk/commands/validate/tests/BA_validators_test.py [hunk 6]
```diff
@@ -2418,3 +2421,73 @@ def test_IsTestsSectionValidValidator_obtain_invalid_content_items():
...
+def test_is_command_or_script_name_starts_with_digit_invalid():
+    """
+    Given
+    - One invalid integration with one command starting with a digit character.
+    When
+    - Calling the IsCommandOrScriptNameStartsWithDigitValidator obtain_invalid_content_items function.
+    Then
+    - Make sure one failure is returned and the error message is correct.
+    """
+    with ChangeCWD(REPO.path):
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/tests/BA_validators_test.py:2435]: not sure you need the ChangeCWD part.

File: demisto_sdk/commands/validate/tests/BA_validators_test.py [hunk 7]
```diff
@@ -2418,3 +2421,73 @@ def test_IsTestsSectionValidValidator_obtain_invalid_content_items():
...
+        content_items = [
+            create_integration_object(
+                paths=["script.commands"],
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/tests/BA_validators_test.py:2438]: let's also add a valid command here and ensure it's not mentioned.

**Review by YuvHayun — CHANGES_REQUESTED** [review]
Looks good!
minor comments.

File: demisto_sdk/commands/validate/tests/BA_validators_test.py (hunk 5 above)
Comment (kamalq97) [reply demisto_sdk/commands/validate/tests/BA_validators_test.py]: You're right! Removed from my unit tests in commit ceaef8a04f169a5a95f9188fd1f16c66ef316110 (and needs to be removed in many others as well)

File: demisto_sdk/commands/validate/tests/BA_validators_test.py (hunk 6 above)
Comment (kamalq97) [reply demisto_sdk/commands/validate/tests/BA_validators_test.py]: You're right! Removed from my unit tests in commit ceaef8a04f169a5a95f9188fd1f16c66ef316110 (and needs to be removed in many others as well)

File: demisto_sdk/commands/validate/tests/BA_validators_test.py (hunk 7 above)
Comment (kamalq97) [reply demisto_sdk/commands/validate/tests/BA_validators_test.py]: Added a valid command alongside the invalid one and made sure it does not appear in the error message

File: demisto_sdk/commands/validate/sdk_validation_config.toml (hunk 2 above)
Comment (kamalq97) [reply demisto_sdk/commands/validate/sdk_validation_config.toml:161]: Added under `[use_git] select` in commit d49bff905078cc4b9794bed4d2d6d59589faf278

**Review by YuvHayun — APPROVED** [review]
Nice!

**Review by JasBeilin — APPROVED** [review]
Great!

**Pull request merged**